nipype/interfaces/mixins/reporting.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+""" class mixin and utilities for enabling reports for nipype interfaces """
+from __future__ import (print_function, division, unicode_literals,
+                        absolute_import)
+
+import os
+from abc import abstractmethod
+
+from ... import logging
+from ..base import (
+    File, traits, BaseInterface, BaseInterfaceInputSpec, TraitedSpec)
+
+iflogger = logging.getLogger('interface')
+
+
+class ReportCapableInputSpec(BaseInterfaceInputSpec):
+    generate_report = traits.Bool(False, usedefault=True,
```

Comment:
As for `terminal_output` and `ignore_errors`, this should not be an input. Should be a state variable of the instance.